### PR TITLE
netutils/dhcpc: Fix dhcpc lease bug

### PR DESCRIPTION
**Pull request by coffee809721232**
## Summary
netutils/dhcpc: Fix dhcpc lease bug

Signed-off-by: songlinzhang songlinzhang@xiaomi.com
## Impact
dhcpc lease management
## Testing
call dhcpc_request_async

### CODE REVIEW

File: netutils/dhcpc/dhcpc.c [hunk 1]
```diff
@@ -458,6 +458,7 @@ static void *dhcpc_run(void *args)
   struct dhcpc_state result;
   int ret;
 
+  memset(&result, 0, sizeof(result));
```
Comment (xiaoxiang781216) [inline netutils/dhcpc/dhcpc.c:461]: why need memset here?

Comment (coffee809721232) [reply netutils/dhcpc/dhcpc.c:461]: If the first request fails, there may be a delay before the second request.

Comment (xiaoxiang781216) [reply netutils/dhcpc/dhcpc.c:461]: If so, should we move into while block to handle the fail in no first loop?

Comment (xiaoxiang781216) [reply netutils/dhcpc/dhcpc.c:461]: @normanr please take a look.

Comment (coffee809721232) [reply netutils/dhcpc/dhcpc.c]: > If so, should we move into while block to handle the fail in no first loop?

Yes, updated

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**